configure.ac

```diff
@@ -100,6 +100,19 @@ if test "$os_unix" = "yes"; then
 		[AC_DEFINE([HAVE_GMTIME_R], 1, [Whether gmtime_r() is available])],
 		[AC_MSG_ERROR([could not find required gmtime_r() function])])
 
+	AC_MSG_CHECKING([whether getpwuid_r() accepts 5 arguments])
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pwd.h>]],
+	                                   [[getpwuid_r(0,0,0,0,0);]])],
+					  [],
+					  [AC_COMPILE_IFELSE(
+					   [AC_LANG_PROGRAM([[#define _POSIX_PTHREAD_SEMANTICS
+#include <pwd.h>]],
+									    [[getpwuid_r(0,0,0,0,0);]])],
+					   [AC_DEFINE([_POSIX_PTHREAD_SEMANTICS], [1],
```

Comment:
Reason I did this, was to ensure the define is added only when it "helps".  I will try using AC_USE_SYSTEM_EXTENSIONS (2.60+) which should define this one, and more.  Hopefully that won't conflict.  Then the entire check can go, I presume.
What do you want to guard in the end?  The extra argument, or the entire call?